### PR TITLE
Clean up - Black Friday Part

### PR DESCRIPTION
**Pull request by yasminyin96**
* Clean up -  Black Friday Part

### How was this tested?
- [ ] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### CODE REVIEW

**Review by yozuysal — APPROVED** [review]
thanks for the clean up!

File: Client/Frontend/ZeroQuery/ZeroQueryModel.swift [hunk 1]
```diff
@@ -172,33 +174,6 @@ class ZeroQueryModel: ObservableObject {
...
-                    )
-                },
-                onClose: {
-                    ClientLogger.shared.logCounter(
-                        .CloseBlackFridayPromo)
-                    Defaults[.seenBlackFridayFollowPromo] = true
-                    self.promoCard = nil
-                })
-        } else if promoCardTypeArm == .previewSignUp && shouldShowDefaultBrowserPromoCard() {
-            promoCard = defaultPromoCard()
```
Comment (chung-lai) [inline Client/Frontend/ZeroQuery/ZeroQueryModel.swift:201]: We should not remove this `else if promoCardTypeArm == .previewSignUp && shouldShowDefaultBrowserPromoCard() {` part. We still show that in the app.

File: Client/Frontend/ZeroQuery/ZeroQueryModel.swift [hunk 2]
```diff
@@ -120,7 +120,9 @@ class ZeroQueryModel: ObservableObject {
         // TODO: refactor the promo card triggering logic to make it easier to customize
         let promoCardTypeArm = NeevaExperiment.arm(for: .promoCardTypeAfterFirstRun)
 
-        if (promoCardTypeArm == .control || promoCardTypeArm == nil)
+        if (promoCardTypeArm == .control
+            || promoCardTypeArm == nil
+            || promoCardTypeArm == .previewSignUp)
```
Comment (chung-lai) [inline Client/Frontend/ZeroQuery/ZeroQueryModel.swift:125]: how does this extra condition `promoCardTypeArm == .previewSignUp` got added?

Comment (yasminyin96) [reply Client/Frontend/ZeroQuery/ZeroQueryModel.swift:125]: The original is:
if (promoCardTypeArm == .control || promoCardTypeArm == nil) && shouldShowDefaultBrowserPromoCard()) {
    promoCard = defaultPromoCard()
} else if ...
...
else if (promoCardTypeArm == .previewSignUp && shouldShowDefaultBrowserPromoCard()) {
   promoCard = defaultPromoCard()
}

Therefore, I combine them into one if statement as
if (promoCardTypeArm == .control || promoCardTypeArm == nil || promoCardTypeArm == .previewSignUp) && shouldShowDefaultBrowserPromoCard()) {}

Comment (chung-lai) [reply Client/Frontend/ZeroQuery/ZeroQueryModel.swift:125]: @EdwardJTL, yes, Yasmin is working on cleaning up the promo card logic by moving it into ViewModel as it's not very readable at the moment with a bunch of `if/else`

@yasminyin96, don't think we can simply combine it. The reason we had that in two different places between we want to maintain the ordering of the condition check :(

Can't wait for some clean up here!

Comment (yasminyin96) [reply Client/Frontend/ZeroQuery/ZeroQueryModel.swift:125]: > @EdwardJTL, yes, Yasmin is working on cleaning up the promo card logic by moving it into ViewModel as it's not very readable at the moment with a bunch of `if/else`
> 
> @yasminyin96, don't think we can simply combine it. The reason we had that in two different places between we want to maintain the ordering of the condition check :(
> 
> Can't wait for some clean up here!

@chung-lai  Make sense, then could I just remove the black friday part first? And I'll look into if/else part later.

Comment (EdwardJTL) [reply Client/Frontend/ZeroQuery/ZeroQueryModel.swift:125]: ah right, i didnt see the other checks in between

**Pull request closed without merging**